src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-tpm2.c

```diff
@@ -0,0 +1,230 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <errno.h>
+#include <libcryptsetup.h>
+
+#include "cryptsetup-token.h"
+#include "cryptsetup-token-util.h"
+#include "hexdecoct.h"
+#include "luks2-tpm2.h"
+#include "memory-util.h"
+#include "tpm2-util.h"
+
+#define TOKEN_NAME "systemd-tpm2"
+#define TOKEN_VERSION_MAJOR "1"
+#define TOKEN_VERSION_MINOR "0"
+
+/* for libcryptsetup debug purpose */
+_public_ const char *cryptsetup_token_version(void) {
+        return TOKEN_VERSION_MAJOR "." TOKEN_VERSION_MINOR;
```

Comment:
Could we have also systemd version here? It would help to detect some old plugin in libcryptsetup log.

Reply:
Ok, so libcryptsetup debug log will print following message for every systemd plugin:

```
# Token handler systemd-tpm2-1.0 systemd-v248 (248-1607-g22a5eb9+) loaded sucessfuly.
```